src/core/openthread.cpp

```diff
@@ -657,118 +683,145 @@ ThreadError otIp6AddressFromString(const char *str, otIp6Address *address)
     return static_cast<Ip6::Address *>(address)->FromString(str);
 }
 
-const otNetifAddress *otGetUnicastAddresses(void)
+const otNetifAddress *otGetUnicastAddresses(otContext *aContext)
 {
-    return sThreadNetif->GetUnicastAddresses();
+    return aContext->mThreadNetif.GetUnicastAddresses();
 }
 
-ThreadError otAddUnicastAddress(otNetifAddress *address)
+ThreadError otAddUnicastAddress(otContext *aContext, otNetifAddress *address)
 {
-    return sThreadNetif->AddUnicastAddress(*static_cast<Ip6::NetifUnicastAddress *>(address));
+    return aContext->mThreadNetif.AddUnicastAddress(*static_cast<Ip6::NetifUnicastAddress *>(address));
 }
 
-ThreadError otRemoveUnicastAddress(otNetifAddress *address)
+ThreadError otRemoveUnicastAddress(otContext *aContext, otNetifAddress *address)
 {
-    return sThreadNetif->RemoveUnicastAddress(*static_cast<Ip6::NetifUnicastAddress *>(address));
+    return aContext->mThreadNetif.RemoveUnicastAddress(*static_cast<Ip6::NetifUnicastAddress *>(address));
 }
 
-void otSetStateChangedCallback(otStateChangedCallback aCallback, void *aContext)
+void otSetStateChangedCallback(otContext *aContext, otStateChangedCallback aCallback, void *aCallbackContext)
 {
-    sNetifCallback.Set(aCallback, aContext);
-    sThreadNetif->RegisterCallback(sNetifCallback);
+    aContext->mNetifCallback.Set(aCallback, aCallbackContext);
+    aContext->mThreadNetif.RegisterCallback(aContext->mNetifCallback);
 }
 
-ThreadError otEnable(void)
-{
-    ThreadError error = kThreadError_None;
+otContext* otEnable(void *aContextBuffer, uint64_t *aContextBufferSize)
```

Comment:
I'm not sure if otEnable/otDisable should be the functions to initialize/uninitialized the otContext, or if we should bring back an otInit (and maybe something like an otUnint). But I really don't understand the point of mEnabled, since generally it always true, unless there is no context, or otDisable has been called, which essentially uninitializes everything; requiring you to reinitialize everything again if you want to use the library.